### DISCUSSION TITLE
I'm curious where the 'help json' is used

### DISCUSSION THREAD

**Discussion by sh-cho** [fluent/fluent-bit#8651] [Q&A] [answered]
Hi, I'm trying to use json-schema of fluent-bit yaml configuration file for auto completion, validation, etc. on IDE.

I know I can generate some helper json file with `-J` option (since #5064), and It is uploaded in packaging step like below.

  - https://packages.fluentbit.io/3.0.0/fluent-bit-schema-3.0.0.json
  - https://packages.fluentbit.io/3.0.0/fluent-bit-schema-pretty-3.0.0.json

So I thought it can maybe used in IDE like below(github workflow yaml json-schema), but didn't work.

<img width="953" alt="Screenshot 2024-03-31 at 1 41 34 AM" src="https://github.com/fluent/fluent-bit/assets/11611397/5c0cef3e-b581-4440-8477-e98320a8f904">
<img width="834" alt="Screenshot 2024-03-31 at 1 41 52 AM" src="https://github.com/fluent/fluent-bit/assets/11611397/4bf450c7-8ee1-4ebc-b7d0-54d1d12cc45e">

Looks like there's standard for json schema.

e.g.
- draft-07: https://json-schema.org/draft-07/json-schema-release-notes
- github action schema: https://github.com/SchemaStore/schemastore/blob/c42ad08e78f70a0d37f6fb53a204371ef80bc3d4/src/schemas/json/github-action.json
  ```json
  {
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "https://json.schemastore.org/github-action.json",
    "$comment": "https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions",
    "additionalProperties": false,
    "definitions": {
      "expressionSyntax": {
        "$comment": "escape `{` and `}` in pattern to be unicode compatible (#1360)",
        "type": "string",
        "pattern": "^\\$\\{\\{(.|[\r\n])*\\}\\}$"
      },
      "stringContainingExpressionSyntax": {
        "$comment": "escape `{` and `}` in pattern to be unicode compatible (#1360)",
        "type": "string",
        "pattern": "^.*\\$\\{\\{(.|[\r\n])*\\}\\}.*$"
      },
      // -- snip --
  }
  ```

So my question is, where does that `-J` generated json file is used?

And Is there a plan for support json schema? If not, I'd love to work on it.

**Comment by patrick-stephens** [comment, marked as answer]
I believe there may be some open issues on the JSON schema....
As to where it is used, it is ingested by various commercial tools - e.g. Calyptia Core.
